locator/token_metadata.cc

```diff
@@ -556,6 +556,9 @@ std::unordered_map<inet_address, host_id> token_metadata_impl::get_endpoint_to_h
     std::unordered_map<inet_address, host_id> map;
     map.reserve(nodes.size());
     for (const auto& [endpoint, node] : nodes) {
+        if (!is_normal_token_owner(endpoint)) {
```

Comment:
Why do you want to restrict to normal token owners?

This function is only used by

```
      {     
         "path":"/storage_service/host_id",
         "operations":[
            {
               "method":"GET",
               "summary":"Retrieve the mapping of endpoint to host ID",
               "type":"array",
               "items":{
                  "type":"mapper"
               },
               "nickname":"get_host_id_map",
               "produces":[
                  "application/json"
               ],    
               "parameters":[
               ]
            }
         ]        
      },
```

Which returns a map of ip <> host_id?

This api does not say it returns only the node that is part of the ring?

For example, when a node is bootstrapping and not part of the ring, it should be returned from the list. 

After this change, the api does not return the node at all.

Reply:
Because it caused a raft test to fail since it returned a decommissioned node with null host_id.

Reply:
But this is not a reason to change an API behavior silently.

Reply:
Good point, will fix.
But I will still restrict the output to nodes with known host_id since the api is about "Retrieve the mapping of endpoint to host ID"